Traceback (most recent call last):
  File "problem.py", line 7, in <module>
    print f()
  File "problem.py", line 3, in f
    y = x
UnboundLocalError: local variable 'x' referenced before assignment
